src/SimpleSAML/Metadata/MetaDataStorageSource.php

```diff
@@ -260,32 +260,35 @@ public function getMetaData(string $entityId, string $set): ?array
      * where the key is the entity id. An empty array may be returned if no matching entities were found.
      * Subclasses should override if their getMetadataSet returns nothing or is slow. Subclasses may want to
      * delegate to getMetaDataForEntitiesIndividually if loading entities one at a time is faster.
-     * @param string[] $entityIds The entity ids to load
-     * @param string $set The set we want to get metadata from.
+     * @param array<string, int> $entityIds        The entity ids to load
+     * @param string             $set              The set we want to get metadata from.
      * @return array An associative array with the metadata for the requested entities, if found.
      */
     public function getMetaDataForEntities(array $entityIds, string $set): array
     {
+        if (empty($entityIds)) {
+            return [];
```

Comment:
I do like this change.  We had an metadata source with a bug that if you loaded 0 entries it would perform super slowly, and this would have help prevent it from triggering.